web/src/app/users/UserNotificationRuleDeleteDialog.js

```diff
@@ -10,21 +10,12 @@ const mutation = gql`
     deleteAll(input: [{ id: $id, type: notificationRule }])
   }
 `
-export default class UserNotificationRuleDeleteDialog extends React.PureComponent {
-  static propTypes = {
-    ruleID: p.string.isRequired,
-  }
-
-  render() {
-    return (
-      <Mutation mutation={mutation} onCompleted={this.props.onClose}>
-        {(commit, status) => this.renderDialog(commit, status)}
-      </Mutation>
-    )
-  }
-
-  renderDialog(commit, { loading, error }) {
-    const { ruleID, ...rest } = this.props
+export default function UserNotificationRuleDeleteDialog({
+  ruleID,
+  onClose,
```

Comment:
this will need to be included in the `...rest` spread so it can properly be passed to the FormDialog. As is, the form can close properly on completion, but the state gets stuck if you cancel deleting and attempt to re-open the form.

it touches a few lines so I can't comment a suggestion, but here's what it'd look like:
```
export default function UserNotificationRuleDeleteDialog(props) {
  const { ruleID, ...rest } = props

  function renderDialog(commit, { loading, error }) {
    return (
      <FormDialog
        title='Are you sure?'
        confirm
        loading={loading}
        errors={nonFieldErrors(error)}
        subTitle='This will delete the notification rule.'
        onSubmit={() => commit({ variables: { id: ruleID } })}
        {...rest}
      />
    )
  }

  return (
    <Mutation mutation={mutation} onCompleted={props.onClose}>
      {(commit, status) => renderDialog(commit, status)}
    </Mutation>
  )
}
```